clang/lib/AST/ASTContext.cpp

```diff
@@ -13462,13 +13462,114 @@ static ElaboratedTypeKeyword getCommonTypeKeyword(const T *X, const T *Y) {
                                             : ElaboratedTypeKeyword::None;
 }
 
+static NestedNameSpecifier *getCommonNNS(ASTContext &Ctx,
```

Comment:
Would LOVE some sort of comments in this function.  it is really quite an adventure to review even now, but I can't imagine coming along this later.  A description and some sort of 'describe-the-flow' comments inline would be really helpful.

Reply:
Done.